contracts/engine/V3/forks/PROHIBITION/GenArt721CoreV3_Engine_Flex_PROHIBITION.sol

```diff
@@ -0,0 +1,2276 @@
+                .additionalPayeePrimarySales;
+        }
+    }
+
+    /**
+     * @notice Returns external asset dependency for project `_projectId` at index `_index`.
+     * If the dependencyType is ONCHAIN, the `data` field will contain the extrated bytecode data and `cid`
+     * will be an empty string. Conversly, for any other dependencyType, the `data` field will be an empty string
+     * and the `bytecodeAddress` will point to the zero address.
+     */
+    function projectExternalAssetDependencyByIndex(
+        uint256 _projectId,
+        uint256 _index
+    ) external view returns (ExternalAssetDependencyWithData memory) {
+        ExternalAssetDependency storage _dependency = projects[_projectId]
+            .externalAssetDependencies[_index];
+        address _bytecodeAddress = _dependency.bytecodeAddress;
+
+        return
+            ExternalAssetDependencyWithData({
+                dependencyType: _dependency.dependencyType,
+                cid: _dependency.cid,
+                bytecodeAddress: _bytecodeAddress,
+                data: (_dependency.dependencyType ==
+                    ExternalAssetDependencyType.ONCHAIN)
+                    ? _bytecodeAddress.readFromBytecode()
+                    : ""
+            });
+    }
+
+    /**
+     * @notice Returns external asset dependency count for project `_projectId` at index `_index`.
+     */
+    function projectExternalAssetDependencyCount(
+        uint256 _projectId
+    ) external view returns (uint256) {
+        return uint256(projects[_projectId].externalAssetDependencyCount);
+    }
+
+    /**
+     * @notice Backwards-compatible (pre-V3) getter returning contract admin
+     * @return address Address of contract admin (same as owner)
+     */
+    function admin() external view returns (address) {
+        return owner();
+    }
+
+    /**
+     * @notice Gets the project ID for a given `_tokenId`.
+     * @param _tokenId Token ID to be queried.
+     * @return _projectId Project ID for given `_tokenId`.
+     */
+    function tokenIdToProjectId(
+        uint256 _tokenId
+    ) public pure returns (uint256 _projectId) {
+        return _tokenId / ONE_MILLION;
+    }
+
+    /**
+     * @notice Convenience function that returns whether `_sender` is allowed
+     * to call function with selector `_selector` on contract `_contract`, as
+     * determined by this contract's current Admin ACL contract. Expected use
+     * cases include minter contracts checking if caller is allowed to call
+     * admin-gated functions on minter contracts.
+     * @param _sender Address of the sender calling function with selector
+     * `_selector` on contract `_contract`.
+     * @param _contract Address of the contract being called by `_sender`.
+     * @param _selector Function selector of the function being called by
+     * `_sender`.
+     * @return bool Whether `_sender` is allowed to call function with selector
```

Comment:
nit: add natspec for param _projectId

Reply:
Not sure why this is still listed as `outdated`. Definitely updated.